File: 3623-count-number-of-trapezoids-i/3623-count-number-of-trapezoids-i.py
MOD = 1000000007
class Solution:
    def countTrapezoids(self, points: List[List[int]]) -> int:
        mp = defaultdict(int)
        for i, j in points:
            mp[j] += 1
        li = [x * (x - 1) // 2 for x in mp.values() if x != 1]
        tl = sum(li)
        res = 0
        for l in li:
            tl -= l
            res += l * tl
        return res % MOD

File: 3623-count-number-of-trapezoids-i/3623-count-number-of-trapezoids-i.java
class Solution {
    private int MOD = 1000000007;
    public int countTrapezoids(int[][] points) {
        int res = 0;
        Map<Integer, Integer> m = new HashMap<>();
        for (int[] pair: points) {
            m.put(pair[1],m.getOrDefault(pair[1],0)+1);
        }
        List<Long> li = m.values().stream().filter(x -> x >= 2).map(x -> ((long)x * (x - 1) / 2)).collect(Collectors.toList());
        long tot = 0;
        for (Long x: li) tot = (tot + x) % MOD;
        for (Long x: li) {
            tot = (tot - x + MOD) % MOD;
            res = (int)(res + (tot * x) % MOD) % MOD;
        }
        return res;
    }
}